pkgs/applications/office/cutemarked-ng/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, qmake
+, qttools
+, qtbase
+, qtwebengine
+, wrapQtAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "CuteMarkEd-NG";
+  version = "unstable-2021-07-29";
+
+  src = fetchFromGitHub {
+    owner = "Waqar144";
+    repo = pname;
+    rev = "9431ac603cef23d6f29e51e18f1eeee156f5bfb3";
+    sha256 = "sha256-BiF63o/XbnUvDBGX2lGTb2Ht589bHcc31UcyXfPhC/c=";
+    fetchSubmodules = true;
```

Comment:
https://github.com/Waqar144/CuteMarkEd-NG/blob/develop/.gitmodules

We have hunspell and md4c already in nixpkgs. Please add qmarkdowntextedit as a library rather than vendoring it.

Reply:
We need patch the following code to remove the dependency on  vendored  library.
And upstream should be reluctant to merge this patch.
```
app/app.pro
43:include(../3rdparty/qmarkdowntextedit/qmarkdowntextedit.pri)
44:include(../3rdparty/hunspell/hunspell.pri)

app/cutemarkdownhighlighter.h
22:#include "../3rdparty/qmarkdowntextedit/markdownhighlighter.h"

test/integration/integration.pro
64:INCLUDEPATH += $$PWD/../../3rdparty/
65:DEPENDPATH += $$PWD/../../3rdparty/

app-static/app-static.pro
16:INCLUDEPATH += $$PWD/../3rdparty/md4c/src
38:    $$PWD/../3rdparty/md4c/src/md4c.c \
39:    $$PWD/../3rdparty/md4c/src/entity.c \
40:    $$PWD/../3rdparty/md4c/src/md4c-html.c
68:    $$PWD/../3rdparty/md4c/src/md4c.h \
69:    $$PWD/../3rdparty/md4c/src/entity.h \
70:    $$PWD/../3rdparty/md4c/src/md4c-html.h

```

Reply:
Thats not that big and I think it is reasonable to have a patch in that size to not vendor libraries.